modin/backends/base/query_compiler.py

```diff
@@ -1569,49 +1585,77 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
-    def get_index_name(self):
+    def get_index_name(self, axis=0):
         """
-        Get index name.
+        Get index name of specified axis.
+
+        Parameters
+        ----------
+        axis: int (default 0),
+            Axis to return index name on.
 
         Returns
         -------
         hashable
             Index name, None for MultiIndex.
         """
-        return self.index.name
```

Comment:
What about column name?

Reply:
`self.get_axis(axis).name` return column name in case of `axis=1`